### PR TITLE
Implement new api endpoint @ogds-group-listing.

### PR DESCRIPTION
**Pull request by elioschmutz**
This PR implements the new `@ogds-group-listing` endpoint and makes adjustments to the already existing `IGroupData` serializer and ogds-group serializer.

The `@ogds-group-listing` endpoint is needed by the fronted to be able to list local groups in a management view.
To be able to fetch additional user-information from plone, we have to provide the `groupurl` for a serialized ogds-group.

Then, a serialized `IGroupData` should provide the users in a json-ld way, not only the userids. This ensures, that we can use the serialized group-data in the gever-ui without any custom implementations for groups.

## `@ogds-groups-listing`
Bildschirmfoto 2020-09-30 um 11 48 16

## Extended ogds-group serializer with `groupurl`
Bildschirmfoto 2020-09-30 um 11 47 31

### Before
Bildschirmfoto 2020-09-30 um 11 49 19

### After
Bildschirmfoto 2020-09-30 um 11 11 01

Jira: https://4teamwork.atlassian.net/browse/NE-83

## Checklist (Must have)

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

## Checklist (if applicable)

_Only applicable should be left and checked._

- API change:
  - [x] Documentation is updated
  - If breaking:
    - [x] api-change label added
    - [ ] #delivery channel notified about breaking change
    - [ ] Scrum master is informed
- New functionality:
  - [ ] for `document` also works for `mail`
  - [ ] for `task` also works for `forwarding`
- Upgrade steps (changes in profile):
  - [ ] Make it deferrable if possible
  - [ ] Execute as much as possible conditionally
- DB-Schema migration
  - [ ] All changes on a model (columns, etc) are included in a DB-schema migration.
  - [ ] Constraint names are shorter than 30 characters (`Oracle`)
- [ ] Change could impact client installations, client policies need to be adapted
- New translations
  - [ ] All msg-strings are unicode
- Change in schema definition:
  - [ ] If `missing_value` is specified, then `default` has to be set to the same value

### CODE REVIEW

**Review by njohner — CHANGES_REQUESTED** [review]
Looks good overall. Just had some minor comments and I'm not sure about the user serialization. Open for discussion

File: opengever/api/serializer.py [hunk 1]
```diff
@@ -196,6 +199,32 @@ def add_additional_metadata(self, data):
...
+    def __call__(self):
+        data = super(GeverSerializeGroupToJson, self).__call__()
+        data['@type'] = self.content_type
+
+        # The user-items are just userids by default which is not the expected
+        # response for summarized users. We have to extend the items manually
+        # with additional metadata to make this endpoint behaves like other ones.
+        user_items = []
+        for userid in data.get('users').get('items'):
+            user_items.append({
```
[the remark this answers is not in the document]
Comment (elioschmutz) [reply opengever/api/serializer.py:217]: The problem here is, that we don't really have a plone user which we can serialize. looking up the user here would cost too much. That's the reason why i do it by hand.

About the `token`. Thats absolutely true. I thought I discussed this with you. It's the way how it is implemented for all other "user" related attributes. But I agree, this is absolutely not the right way. I think we really have to handle this in the frontend. Even that we have to always return the serialized user needs to be fixed as soon as possible (when not now) with a central actor lookup endpoint and frontend-side caching.

File: docs/public/dev-manual/api/listings.rst [hunk 2]
```diff
@@ -373,3 +373,85 @@ Mit ``filters.last_login:record:list`` kann nach dem Zeitpunkt des letzten Login
...
+
+Auflistung der OGDS-Gruppen
+===========================
+
+Mit den Endpoints ``@ogds-group-listing`` können Gruppen aus dem ogds aufgelistet werden.
+Diese Endpoints liefern inhaltlich die gleiche Struktur wie der ``@listing`` Endpoint, unterstützen
+aber nur ein Subset der Parameter. Im Moment ist es nicht möglich die
+``columns`` anzugeben, sondern es werden immer alle vom Modell
+untertstützten Attribute zurückgegeben. Des weiteren ist der ``depth``
+Paremeter nicht implementiert, ``facets`` werden ebenfalls nicht unterstützt.
```
[the remark this answers is not in the document]
Comment (elioschmutz) [reply docs/public/dev-manual/api/listings.rst:386]: I just copied it from the `@ogds-user-listing` endpoint. But you're right. I removed it.

File: opengever/api/tests/test_ogdsgrouplisting.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+        ogds_service().fetch_group('projekt_a').is_local = True
+
+        ogds_service().fetch_group('projekt_b').is_local = None
+        ogds_service().fetch_group('projekt_laeaer').is_local = False
+
+        browser.open(self.contactfolder,
+                     view='@ogds-group-listing?filters.is_local:record:boolean=False',
+                     headers=self.api_headers)
+
```
[the remark this answers is not in the document]
Comment (elioschmutz) [reply opengever/api/tests/test_ogdsgrouplisting.py:152]: 👍

File: opengever/api/serializer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (elioschmutz) [reply opengever/api/serializer.py:217]: @njohner the story about fixing the lookup mess is here: https://4teamwork.atlassian.net/browse/CA-664

So I would say we merge this PR as it is. We have to make a breaking api change PR anyway for the `CA-664`. I'll list this PR in the story `CA-664` to make sure we don't forget to revert it.

Do you agree with that?

I updated the PR with all your other change requests. So could you take another look?

[the remark this answers is not in the document]
Comment (njohner) [reply opengever/api/serializer.py:217]: Fine with me, thanks for creating the story.

**Review by njohner — APPROVED** [review]
LGTM, except that tests are failing ;-)

**Pull request merged**